musa: remove nrows_x in mul_mat_q_process_tile

yeahdongcn opened the pull request:
*Make sure to read the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md) before submitting a PR*

Fix a new warning detected by our internal CI.

JohannesGaessler commented:
Previously `nrows_x` was needed to determine the maximum extent of valid memory. I moved that one level higher and forgot to remove the argument in `mul_mat_q_process_tile`. So it should simply be removed since it's not used for anything (create a new commit from master to avoid a merge conflict).

yeahdongcn committed:
Remove nrows_x in mul_mat_q_process_tile

Signed-off-by: Xiaodong Ye <xiaodong.ye@mthreads.com>

yeahdongcn commented:
> Previously `nrows_x` was needed to determine the maximum extent of valid memory. I moved that one level higher and forgot to remove the argument in `mul_mat_q_process_tile`. So it should simply be removed since it's not used for anything (create a new commit from master to avoid a merge conflict).

Thanks for the review! I've updated the PR based on your comments.
